include/RAJA/policy/atomic_builtin.hpp

```diff
@@ -42,6 +42,19 @@ namespace detail
 
 #if defined(RAJA_COMPILER_MSVC) || (defined(_WIN32) && defined(__INTEL_COMPILER))
 
+RAJA_DEVICE_HIP
+RAJA_INLINE unsigned builtin_atomic_load(unsigned volatile *acc)
+{
+  return RAJA::util::reinterp_A_as_B<long, unsigned>(_InterlockedOr((long volatile *)acc, 0));
```

Comment:
We shouldn't be using volatile, shall we cast it away here?

Reply:
So you're suggesting the following?

`return RAJA::util::reinterp_A_as_B<long, unsigned>(_InterlockedOr((long *)acc, 0));`

Reply:
Or should builtin_atomic_load not be taking a volatile pointer?

Reply:
This should also use unsigned

Reply:
We should remove volatile through the whole atomic call stack, but we can do that later.

Reply:
Microsoft does not support unsigned.

Reply:
I added a static_assert to check if they are the same types. I also removed volatile on the set of builtin_atomic_load functions.

Reply:
I would leave the volatile on the builtin_atomic_load functions for now so they are easy to call from other atomics, just cast it away internally.

Reply:
Huh. Adding volatile back in causes the tests to fail.

Reply:
Sound like we have some volatility issues.... 😄